Core/BossPrototype.lua

```diff
@@ -1464,6 +1464,8 @@ end
 -- @param[opt] icon the bar icon (spell id or texture name)
 function boss:Bar(key, length, text, icon)
 	local textType = type(text)
+	local bars = core:GetPlugin("Bars")
+	if bars and not bars.db.profile["cast"] and text == L.cast then return end
```

Comment:
Yes, per-boss options would be required which is why this isn't a simple feature. It requires a new API implementation and propagation across all modules.